describegroups.go

```diff
@@ -174,3 +313,119 @@ func (t *describeGroupsResponseV0) readFrom(r *bufio.Reader, sz int) (remain int
 
 	return
 }
+
+// decodeMemberMetadata converts raw metadata bytes to a
+// DescribeGroupsResponseMemberMetadata struct.
+//
+// See https://github.com/apache/kafka/blob/2.4/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocol.java#L49
+// for protocol details.
+func decodeMemberMetadata(rawMetadata []byte) (DescribeGroupsResponseMemberMetadata, error) {
+	mm := DescribeGroupsResponseMemberMetadata{}
+
+	if len(rawMetadata) == 0 {
+		return mm, nil
+	}
+
+	buf := bytes.NewBuffer(rawMetadata)
+	bufReader := bufio.NewReader(buf)
```

Comment:
you can use the `bytes.Buffer` returned from `bytes.NewBuffer` as a reader below, unless you wanted the buffering here

Reply:
Alternatively, `bytes.Reader` with `bytes.NewReader(rawMetadata)` could be an interesting option.

Reply:
I think the `read...` helpers we use below specifically require a `bufio.Reader` as opposed to a generic `io.Reader` since they're peeking ahead. Let me know if I'm missing something here?

Reply:
You are correct 👍

Reply:
Ok :-).